pkg/parser/yaml/parser.go

```diff
@@ -65,7 +66,41 @@ func (p *Parser) GetKind() model.FileKind {
 	return model.KindYAML
 }
 
-func playbookParser(fileContent []byte) ([]model.Document, error) {
+func addExtraInfo(documents []model.Document, filePath string) []model.Document {
+	var elements map[string]interface{}
+	var certInfo map[string]interface{}
+	var swaggerInfo map[string]interface{}
+
+	for _, documentPlaybooks := range documents { // iterate over documents
+		for _, resources := range documentPlaybooks["playbooks"].([]interface{}) { // iterate over playbooks
+			for _, v := range resources.(map[string]interface{}) {
+				_, ok := v.(map[string]interface{})
+				if ok {
+					elements = v.(map[string]interface{})
+					if elements["certificate"] != nil {
+						content := additional.CheckCertificate(elements["certificate"].(string))
+						if content != "" {
+							certInfo = additional.AddCertificateInfo(filePath, content)
+							if certInfo != nil {
+								elements["certificate"] = certInfo
+							}
+						}
+					}
+					if elements["swagger_file"] != nil {
+						swaggerInfo = additional.AddSwaggerInfo(filePath, elements["swagger_file"].(string))
+						if swaggerInfo != nil {
+							elements["swagger_file"] = swaggerInfo
+						}
+					}
+				}
+			}
+		}
```

Comment:
Extract nested for loops to separated functions passing down parameters and returning values as needed. I could only find an example of this in Python https://codereview.stackexchange.com/a/191823

Reply:
Thank you so much, Rogério!